File: frontend/src/api/employee.ts
import request from '@/utils/request';

export interface EmployeeInfo {
  id: number;
  employeeNo: string;
  name: string;
  gender?: number; // 0-女，1-男
  age?: number;
  phone?: string;
  email?: string;
  departmentId: number;
  departmentName?: string;
  position?: string;
  entryDate?: string;
  status: number; // 0-离职，1-在职
  createTime?: string;
  updateTime?: string;
}

export interface EmployeeQueryParams {
  name?: string;
  employeeNo?: string;
  departmentId?: number;
  status?: number;
  pageNum?: number;
  pageSize?: number;
}

export interface EmployeeSaveParams {
  id?: number;
  employeeNo: string;
  name: string;
  gender?: number;
  age?: number;
  phone?: string;
  email?: string;
  departmentId: number;
  position?: string;
  entryDate?: string;
  status: number;
}

// 获取员工列表（分页）
export const getEmployeeList = (params: EmployeeQueryParams) => {
  return request.get<{
    list: EmployeeInfo[];
    total: number;
  }>('/employee/list', { params });
};

// 获取员工详情
export const getEmployeeById = (id: number) => {
  return request.get<EmployeeInfo>(`/employee/${id}`);
};

// 创建员工
export const createEmployee = (data: EmployeeSaveParams) => {
  return request.post<void>('/employee', data);
};

// 更新员工
export const updateEmployee = (id: number, data: EmployeeSaveParams) => {
  return request.put<void>(`/employee/${id}`, data);
};

// 删除员工
export const deleteEmployee = (id: number) => {
  return request.delete<void>(`/employee/${id}`);
};

File: backend/src/main/java/com/zssystem/entity/Employee.java
package com.zssystem.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("employee")
public class Employee {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("employee_no")
    private String employeeNo;

    private String name;

    private Integer gender; // 0-女，1-男

    private Integer age;

    private String phone;

    private String email;

    @TableField("department_id")
    private Long departmentId;

    private String position;

    @TableField("entry_date")
    private LocalDate entryDate;

    private Integer status; // 0-离职，1-在职

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    private Integer deleted;
}
